### PR TITLE
Upgrade to Socket io 1.3

### PR DESCRIPTION
**Pull request by Dignifiedquire**
This is very much a work in progress, but from my tests neither the PR in #1257 nor #1258 does actually work. I hope to finish this in the next couple of days.

**Update1:** This only works when manually pulling in the forked & built version of `karma-jasmine@0.1.5` that's why travis will be complaining, until we updated the release there.

**Update2:** Finally all tests passing locally :)

**Update3:** Update 1 is now obsolete as I switched the client tests to mocha.

### CODE REVIEW

**Comment by maksimr** [comment]
@Dignifiedquire Awesome!! :)

**Comment by Dignifiedquire** [comment]
@maksimr and everyone else, basics are working for me, now I need some more eyes on this to test it out please.

**Comment by maksimr** [comment]
@Dignifiedquire check in our projects all work fine with latest version of `karma-jasmine`, `karma-mocha`, `karma-webpack` ;)

**Comment by Dignifiedquire** [comment]
@maksimr great. Also I was tired of jasmine so I switched the client tests to run on mocha with browserify :)

**Comment by maksimr** [comment]
**LGTM**

**Comment by Dignifiedquire** [comment]
@nikku @bendrucker do you have any ideas why this is failing? All is running fine on my local machine (osx + FF) but it seems travis doesn't like the setup :(

**Comment by bendrucker** [comment]
karma-browserify has a hack to add a preprocessor for the bundle file: 

https://github.com/nikku/karma-browserify/blob/5d75d41879a32cc5a8b62b7c821b1ddbd796e870/lib/preprocessor.js#L3
https://github.com/nikku/karma-browserify/blob/5d75d41879a32cc5a8b62b7c821b1ddbd796e870/lib/preprocessor.js#L35

I'm betting the Karma under test !== the Karma that gets required on line 2. Try manually implementing the result from line 35 when you create your Karma config, adding: `'**/*': ['browserify-bundle']` to `config.preprocessors`.

**Comment by Dignifiedquire** [comment]
Thanks a lot @bendrucker that fixed it :)

**Comment by bendrucker** [comment]
Awesome. I'm guessing there's a reason npm doesn't handle that peer dep more cleverly. Couldn't find it though.

**Comment by Dignifiedquire** [comment]
@bendrucker looks like that wasn't actually the issue, travis started failing after one successful run again. The issue is that we are deleting `node_modules/karma` but didn't symlink it, so `karma-browserify` couldn't find any installation of karma.

**Comment by Dignifiedquire** [comment]
@maksimr need a new review, needed more fixes for tests including https://github.com/karma-runner/integration-tests/pull/4 to fix the integration tests.

**Comment by lygstate** [comment]
it's better not use 1.3.5, use 1.3 should be better, cause only the master 
compat with iojs 2.1.0

**Comment by lygstate** [comment]
Ping for review and pull

**Comment by Dignifiedquire** [comment]
Merged as 79072aefee2a70d0fd53a47f159e455297b0bfdc

File: lib/launchers/process.js [hunk 1]
```diff
@@ -64,11 +64,11 @@ var ProcessLauncher = function(spawn, tempDir, timer) {
 
     var errorOutput = '';
 
-    self._process.on('close', function(code) {
+    self._process.on('exit', function(code) {
       self._onProcessExit(code, errorOutput);
     });
 
-    self._process.on('error', function(err) {
+    self._process.on('karma_error', function(err) {
```
Comment (badcorporatelogo) [inline lib/launchers/process.js]: This line does not look correct. The error event in this case is coming from the child process, indicating that spawn failed. It is different from the error event emitted by the socket, which was renamed from "error" to "karma_error".

Comment (Dignifiedquire) [inline lib/launchers/process.js]: You are totally right, thanks a lot fixed in https://github.com/karma-runner/karma/commit/45a69221703bbd043cd71b8b0934c4be5339e111

**Pull request closed without merging**